ompi/mca/coll/base/coll_base_alltoallv.c

```diff
@@ -281,8 +281,10 @@ ompi_coll_base_alltoallv_intra_basic_linear(const void *sbuf, const int *scounts
         for( i = 0; i < nreqs; i++ ) {
             if (MPI_REQUEST_NULL == reqs[i]) continue;
             if (MPI_ERR_PENDING == reqs[i]->req_status.MPI_ERROR) continue;
-            err = reqs[i]->req_status.MPI_ERROR;
-            break;
+            if (reqs[i]->req_status.MPI_ERROR != MPI_SUCCESS) {
```

Comment:
to follow the convention, I think the code should be
`if (MPI_SUCCESS == reqs[i]->req_status.MPI_ERROR) continue;`

Reply:
No, you don't need to make the logic opposite what it should be just to adhere to "constants on the left of ==".

If the natural test is `!=`, then use `!=`.

Also, please put `{}` around all blocks; even 1-line blocks.  Apparently whoever wrote the original code here was a Bad Person and did `if (foo) continue;`.  😢

Reply:
Other collectives (alltoall, barrier, bcast, gather, reduce, scatter) had the similar error handling, so apply the change for them as well. I have to callout that I don't a reproducer for other collectives, but I think adding a check to ensure an error code is returned does not hurt.

For those `if () continue;` above my change, I can add a separate commit in this PR to fix them.

Reply:
Pushed the change.